funding/manager.go

```diff
@@ -3808,23 +3821,33 @@ func (f *Manager) waitForZeroConfChannel(c *channeldb.OpenChannel) error {
 				"six confirmations: %v", err)
 		}
 
-		// TODO: Make this atomic!
-		ourPolicy, err := f.cfg.DeleteAliasEdge(c.ShortChanID())
+		// The underlying graph entry for this channel id needs to be
+		// reassigned with the new confirmed scid. Moreover channel
+		// updates with the alias scid are removed so that we do not
+		// relay them to the broader network.
+		ourPolicy, err := f.cfg.ReAssignSCID(
```

Comment:
I don't think this fixes the issue mentioned in the PR description re the possible race. We can still receive an `channel_update` from the peer while we are in the process of re-assigning the SCID. Meanwhile, in the gossiper, if we receive a `channel_update` that we don't understand, it's put in the `prematureChannelUpdates` and will be processed later.

Reply:
I described the problem in detail here: https://github.com/lightningnetwork/lnd/pull/7759/commits/ef6cd7a96fb66680b5d3037c8761c4b41ecfec6e

Looking back I think this issue could have been solved by deleting the rejectCache before adding the edge via the `addToGraph` function. Tho I think having the deletion and addition atomic is the better desing wdyt ?